webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTableReducer.js

```diff
@@ -0,0 +1,47 @@
+import Immutable from 'seamless-immutable';
+import {
+  TABLE_REQUEST,
+  TABLE_SUCCESS,
+  TABLE_FAILURE,
+  INIT_TABLE,
+} from './ModelsTableConsts';
+import { STATUS } from '../../constants';
+
+const initState = Immutable({
+  page: '1',
+  perPage: '20',
+  sortBy: '',
+  sortOrder: '',
+  results: [],
+  error: null,
+  status: STATUS.PENDING,
+  searchQuery: null,
+});
+export default (state = initState, action) => {
+  switch (action.type) {
+    case INIT_TABLE:
+      return Immutable.merge(state, {
+        ...action.payload,
```

Comment:
IMO, better to destruct the payload and be in control about what's get into the store.